third_party/envtest/envtest.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envtest
+
+import (
+	"context"
+	"os"
+
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+// SetupCRDs is provided as a helper env.Func handler that can be used to setup the CRDs that are required
+// to process your controller code for testing. This handlers provides a functionality similar to the one
+// provided by the envtest framework via the `CRDDirectoryPaths` argument of the envtest helper.
+//
+// The crdPath should be the base directory containing all the CRDs that needs to be pre-installed into
+// your cluster before your test handlers are kicked off. This can be integrated with the setup handlers
+func SetupCRDs(crdPath string) env.Func {
+	return func(ctx context.Context, c *envconf.Config) (context.Context, error) {
+		r, err := resources.New(c.Client().RESTConfig())
+		if err != nil {
+			return ctx, err
+		}
+		err = decoder.DecodeEachFile(
+			ctx, os.DirFS(crdPath), "*", decoder.CreateHandler(r),
```

Comment:
If yaml files that are not CRDs are present in the directory, this will attempt to create those as well. This may not be expected -- I think we may want an additional decoder func (or option, TBD) to only apply the "handler" function on a given type.

Reply:
I added a simple Pre Handler filter function option to the newly created method to retain the backward compatibility. We can decided how do we enable this for existing APIs that we already released into the wild